docs/proposals/20220228-advanced-cpuset-manger.md

```diff
@@ -0,0 +1,94 @@
+---
+title: Advanced CPUSet Manager
+authors:
+  - "@szy441687879"
+reviewers:
+  - "@yan234280533"
+  - "@mfanjie"
+creation-date: 2022-02-28
+last-updated: 2022-02-28
+status: provisional
+---
+
+# Advanced CPUSet Manager
+
+- Provide a new mechanism to manage cpuset bypass, to do extra things that kubelet cannot do, such as revise cpuset when pod running, share the cpuset assigned with others and relax restrictions of cpuset allocation.
+
+## Table of Contents
+
+<!-- TOC -->
+
+- [Advanced CPUSet Manager](#advanced-cpuset-manager)
+    - [Table of Contents](#table-of-contents)
+    - [Motivation](#motivation)
+        - [Goals](#goals)
+        - [Non-Goals/Future Work](#non-goalsfuture-work)
+    - [Proposal](#proposal)
+        - [Relax restrictions of cpuset allocation](#relax-restrictions-of-cpuset-allocation)
+        - [Add new annotation to describe the  requirement of cpuset contorl manger](#add-new-annotation-to-describe-the--requirement-of-cpuset-contorl-manger)
+        - [Advanced CPU Manager component](#advanced-cpu-manager-component)
+        - [User Stories](#user-stories)
+            - [Story 1](#story-1)
+            - [Story 2](#story-2)
+        - [Risks and Mitigations](#risks-and-mitigations)
+
+<!-- /TOC -->
+## Motivation
+Some latency-sensitive application have lower lantency and cpu usage when running with specific cores, which results in fewer context switchs and higer cache affinity.
+But kubelet will always exclude assigned cores in shared cores, which may waste resources.Offline and other online pods can running on the codes actually. In our experiment, for the most part, it is barely noticeable for performance of service.
+
+### Goals
+
+- Provide a new mechanism to manage cpuset bypass
+- Provide a new cpuset control way "shared"
+- Allow revise cpuset when pod running
+- Relax restrictions of cpuset allocation
+
+
+### Non-Goals/Future Work
+
+- Solve the conflict with kubelet static cpuset control (need close)
```

Comment:
conflicts

Reply:
May I ask what is `need close` means? thanks

Reply:
It means if you set cpu-manager-policy=static in kubelet, it will be conflicted with crane-agent.You need to close the static  cpu-manager-policy in kubelet.
Sorry, it's not clear here.